60_networks.Rmd

```diff
@@ -0,0 +1,855 @@
+
+-   Aggregation to genus level
+-   Add relative abundance assay
+-   Prevalence filtering (keep genera with prevalence \> 20%)
+-   Add assay with clr-transformed abundances
+
+```{r}
+# Agglomerate to genus level
+tse_genus <- agglomerateByRank(tse0, rank = "genus") 
+
+# Add relative abundances
+#debugonce(vegan::decostand)
+tse <- transformAssay(tse_genus,
+                       assay.type = "counts",
+                       method = "relabundance",
+                       MARGIN = "samples") 
+
+# Filter by prevalence
+tse <- subsetByPrevalentFeatures(tse,
+                                 detection = 0,
+                                 prevalence = 0.2,
+                                 assay.type = "relabundance")
+
+
+# Add clr-transformed abundances
+tse <- transformAssay(tse, method="clr", pseudocount=1)
+
+dim(tse0)
+dim(tse)
+```
+
+### SparCC correlation
+
+We start by estimating a sparse correlation matrix using SparCC.
+
+```{r, eval=FALSE}
+sparcc_cor <- SpiecEasi::sparcc(t(assay(tse, "counts")))$Cor
+rownames(sparcc_cor) <- colnames(sparcc_cor) <- rownames(tse)
+```
+
+Two threshold values are used to see the effect of sparsification later in the network plot.
+
+```{r}
+sparcc_cor03 <- sparcc_cor
+
+# Sparsification (threshold = 0.3)
+sparcc_cor03[abs(sparcc_cor03) < 0.3] <- 0
+
+# Compute dissimilarity matrix
+sparcc_diss03 <- sqrt(0.5 * (1 - sparcc_cor03))
+
+# Dissimilarities between nodes with zero correlation are set to 1
+# (these nodes are unconnected and thus have maximum dissimilarity).
+sparcc_diss03[sparcc_cor03 == 0] <- 1
+
+# Compute similarity matrix
+sparcc_sim03 <- 1 - sparcc_diss03
+
+# Turn into graph
+sparcc_graph03 <- adj2igraph(sparcc_sim03)
+
+# Repeat with threshold = 0.4
+sparcc_cor04 <- sparcc_cor
+sparcc_cor04[abs(sparcc_cor04) < 0.4] <- 0
+sparcc_diss04 <- sqrt(0.5 * (1 - sparcc_cor04))
+sparcc_diss04[sparcc_cor04 == 0] <- 1
+sparcc_sim04 <- 1 - sparcc_diss04
+sparcc_graph04 <- adj2igraph(sparcc_sim04)
+```
+
+### Shrinkage proportionality
+
+Since there is no R package implementing the shrinkage proportionality estimator as suggested by [@badri2020shrinkage], we use the `rho_shrink_est()` function provided in the GitHub repository associated with the paper. The function is slightly modified to expect normalized counts as input.
+
+```{r}
+# norm_counts: clr-transformed count matrix with samples in rows
+rho_shrink_est <- function(norm_counts, ...) {
+  shrunk_cov <- cov.shrink(norm_counts, ...)
+  p <- ncol(norm_counts)
+  J <- matrix(rep(diag(shrunk_cov), p), p)
+  rho <- 2 * shrunk_cov / (J + t(J))
+  (rho + t(rho)) / 2
+}
+```
+
+```{r}
+# Apply the shrinkage proportionality estimator to the clr-transformed counts
+prop_est <- as(rho_shrink_est(t(assay(tse, "clr"))), "matrix")
+```
+
+A threshold of 0.4 is used to sparsify the estimated proportionality matrix.
+
```

Comment:
All the methods used in the examples are explained in the introduction of this chapter below the flowchart. We thought it would be better to have a section at the beginning that introduces the methods and puts them in context, so that the examples don't need further explanation. In the introduction, we could add cross-links to the examples so it's clearer which example belongs to which method. Or would you prefer to have the explanations above each application?